pyiceberg/table/inspect.py

```diff
@@ -645,10 +645,10 @@ def data_files(self, snapshot_id: Optional[int] = None) -> "pa.Table":
     def delete_files(self, snapshot_id: Optional[int] = None) -> "pa.Table":
         return self._files(snapshot_id, {DataFileContent.POSITION_DELETES, DataFileContent.EQUALITY_DELETES})
 
-    def all_manifests(self) -> "pa.Table":
+    def all_manifests(self, snapshots: Optional[list[Snapshot]] = None) -> "pa.Table":
         import pyarrow as pa
 
-        snapshots = self.tbl.snapshots()
+        snapshots = snapshots or self.tbl.snapshots()
         if not snapshots:
```

Comment:
Another case of me treating snapshots and snapshot_id's the same... happy to enforce this being snapshot_id's instead

Reply:
Let's save that for another PR. I don't think we can just change this API since folks might be using this.. We could allow for an `Union[list[snapshot], iterable[int]]`?

Reply:
Modified this, let me know what you think